packages/syntax-icons/src/generate-icons/generateIcons.ts

```diff
@@ -0,0 +1,80 @@
+import { promises as fs } from "fs";
+import { transform } from "@svgr/core";
+import path from "path";
+import iconTemplate from "./iconTemplate";
+import { exec } from "child_process";
+
+const __dirname = path.resolve();
+
+const BASE_SVG_PATH = path.join(__dirname, "packages", "syntax-icons", "svgs");
+const EXPORT_PATH = path.join(
+  __dirname,
+  "packages",
+  "syntax-icons",
+  "src",
+  "icons",
+);
+
+function snakeToPascal(str: string): string {
+  return str
+    .toLowerCase()
+    .replace(/([-_][a-z])/g, (group) =>
+      group.toUpperCase().replace("-", "").replace("_", ""),
+    )
+    .replace(/^[a-z]/, (group) => group.toUpperCase());
+}
+
+function getFileName(file: string) {
+  return snakeToPascal(file.replace(".svg", ""));
+}
+
+async function generateIcons() {
+  try {
```

Comment:
@somethiiing do we even need the `try` / `catch` if we just do `throw error` in the `catch`?